src/shared/nsflags.c

```diff
@@ -40,18 +41,40 @@ int namespace_flags_from_string(const char *name, unsigned long *ret) {
 }
 
 int namespace_flags_to_string(unsigned long flags, char **ret) {
+        _cleanup_strv_free_ char **l = NULL;
         _cleanup_free_ char *s = NULL;
+        int r;
+
+        r = namespace_flags_to_strv(flags, &l);
+        if (r < 0)
+                return r;
+
+        s = strv_join(l, NULL);
+        if (!s)
+                return -ENOMEM;
+
+        if (ret)
+                *ret = TAKE_PTR(s);
+
+        return 0;
+}
+
+int namespace_flags_to_strv(unsigned long flags, char ***ret) {
+        _cleanup_strv_free_ char **s = NULL;
         unsigned i;
+        int r;
 
         for (i = 0; namespace_info[i].proc_name; i++) {
                 if ((flags & namespace_info[i].clone_flag) != namespace_info[i].clone_flag)
                         continue;
 
-                if (!strextend_with_separator(&s, " ", namespace_info[i].proc_name))
-                        return -ENOMEM;
+                r = strv_extend(&s, namespace_info[i].proc_name);
+                if (r < 0)
+                        return r;
         }
 
-        *ret = TAKE_PTR(s);
+        if (ret)
+                *ret = TAKE_PTR(s);
```

Comment:
Please add `assert(ret)` to the start of this function, it does not make sense for `ret` to be optional here since the entire goal of the function is to convert the flags to a strv.